src/test/conditions/PrefixSha256_test.cpp

```diff
@@ -0,0 +1,450 @@
+            }
+
+            // The rest should fail:
+            BEAST_EXPECT (! validate (f1, f2.condition(), {}));
+            BEAST_EXPECT (! validate (f2, f1.condition(), {}));
+
+            for (std::size_t i = 1; i != sizeof(msg); ++i)
+            {
+                BEAST_EXPECT (! validate (f1, f2.condition(),
+                    Slice{msg, i}));
+                BEAST_EXPECT (! validate (f2, f1.condition(),
+                    Slice{msg, i}));
+            }
+        }
+
+        {
+            PrefixSha256 f1;
+            f1.setPrefix(makeSlice (prefix1));
+            f1.setSubfulfillment(std::make_unique<PreimageSha256> (
+                makeSlice (prefix1)));
+
+            PrefixSha256 f2;
+            f2.setPrefix(makeSlice (prefix2));
+            f2.setSubfulfillment(std::make_unique<PreimageSha256> (
+                makeSlice (prefix2)));
+
+            BEAST_EXPECT (f1 != f2);
+            BEAST_EXPECT (f1.condition() != f2.condition());
+            BEAST_EXPECT (validate (f1, f1.condition(), {}));
+            BEAST_EXPECT (validate (f2, f2.condition(), {}));
+            BEAST_EXPECT (! validate (f1, f2.condition(), {}));
+            BEAST_EXPECT (! validate (f2, f1.condition(), {}));
+
+            // For preimage conditions, the message shouldn't
+            // matter, so verify that it does not:
+            for (std::size_t i = 1; i != sizeof(msg); ++i)
+            {
+                BEAST_EXPECT (validate (f1, f1.condition(),
+                    Slice(msg, i)));
+                BEAST_EXPECT (validate (f2, f2.condition(),
+                    Slice(msg, i)));
+                BEAST_EXPECT (! validate (f1, f2.condition(),
+                    Slice(msg, i)));
+                BEAST_EXPECT (! validate (f2, f1.condition(),
+                    Slice(msg, i)));
+            }
+        }
+
+        {
+            std::array<std::uint8_t, 32> const sk =
+            {{
+                0x50, 0xd8, 0x58, 0xe0, 0x98, 0x5e, 0xcc, 0x7f,
+                0x60, 0x41, 0x8a, 0xaf, 0x0c, 0xc5, 0xab, 0x58,
+                0x7f, 0x42, 0xc2, 0x57, 0x0a, 0x88, 0x40, 0x95,
+                0xa9, 0xe8, 0xcc, 0xac, 0xd0, 0xf6, 0x54, 0x5c
+            }};
+
+            PrefixSha256 f1;
+            f1.setPrefix(makeSlice (prefix1));
+            f1.setSubfulfillment(std::make_unique<Ed25519> (
+                SecretKey{ sk }, makeSlice (prefix1)));
+
+            PrefixSha256 f2;
+            f2.setPrefix(makeSlice (prefix2));
+            f2.setSubfulfillment(std::make_unique<Ed25519> (
+                SecretKey{ sk }, makeSlice (prefix2)));
+
+            BEAST_EXPECT (f1 != f2);
+            BEAST_EXPECT (f1.condition() != f2.condition());
+            BEAST_EXPECT (validate (f1, f1.condition(), {}));
+            BEAST_EXPECT (validate (f2, f2.condition(), {}));
+            BEAST_EXPECT (! validate (f1, f2.condition(), {}));
+            BEAST_EXPECT (! validate (f2, f1.condition(), {}));
```

Comment:
We need a test that validates with a signing subcondition and a non-empty initial message (you said you added one, did I miss it?)